ert_shared/ensemble_evaluator/entity/snapshot.py

```diff
@@ -61,38 +65,53 @@ def update_stage(
         status=None,
         start_time=None,
         end_time=None,
-        queue_state=None,
     ):
-        real = self.update_real(real_id)
-        if stage_id not in real["stages"]:
-            real["stages"][stage_id] = {"steps": {}}
-        stage = real["stages"][stage_id]
+        stage = {}
 
         if status is not None:
             stage["status"] = status
         if start_time is not None:
             stage["start_time"] = start_time
         if end_time is not None:
             stage["end_time"] = end_time
-        if queue_state is not None:
-            stage["queue_state"] = queue_state
-        return stage
+
+        self._apply_update({"reals": {real_id: {"stages": {stage_id: stage}}}})
+
+        if status == "Finished" and self._snapshot.all_stages_finished(real_id):
+            self.update_real(real_id, queue_state="JOB_QUEUE_SUCCESS")
+        if status == "Failed":
+            self.update_real(real_id, queue_state="JOB_QUEUE_FAILED")
+        if status == "Pending":
+            self.update_real(real_id, queue_state="JOB_QUEUE_RUNNING")
+
```

Comment:
You should move this into the prefect ensemble, because the legacy ensemble produces real statuses.